setup.py

```diff
@@ -1,4 +1,12 @@
 from setuptools import setup
 
+# version.py is created when building with SCons.
+# Otherwise, we create it ourselves.
+version = "0.1.0"
+with open("./python/lsst/alert/packet/version.py", "w") as f:
```

Comment:
I don't understand why this is preferable to checking in this file, or informing SCons of the version from setup.cfg. Why do it this way?

Informing SCons from setup.cfg would be nice because it would mean that the version lives in just one file instead of two; the setup.cfg file would be authoritative. It looks like this should be possible from looking at
 - https://github.com/lsst/sconsUtils/blob/master/python/lsst/sconsUtils/scripts.py#L86-L88
and 
 - https://github.com/lsst/sconsUtils/blob/master/python/lsst/sconsUtils/scripts.py#L75-L77

Reply:
> Why do it this way?

Because it's lifted directly from astro_metadata_translator, which is the nearest we have to precedent, and means I was fairly confident that it'd do the Right Thing™. If you think it's inelegant... I don't disagree, and I'm happy to spend a few more minutes looking at it (but would rather not go down a rabbit hole).

Reply:
I did it this way because astro_metadata_translator runs with both setup.py and scons and so I was trying to match the sconsUtils approach in my setup.py. The version is the pip version so yes when scons runs you see a github SHA1 derived version. I didn't know how to fix that discrepancy at the time.

Reply:
I see. I think it's very important for there to be just one version number. I don't really want to have to think about whether `w.2020.25-7-g6c29e90` corresponds to `0.1.0` or whatever.

Either that means that SCons should be in charge, or setuptools should be in charge, but I don't think we should let each use its own versioning scheme, especially not writing to the same file on disk depending on whether `setup.py` has been interpreted before or after the last run of `scons`.

Reply:
I'm a bit confused because this package no longer uses Scons so it is not required to do what sconsUtils does. astro_metadata_translator does use scons and pip so has to create a version.py.   Whatever you do, you still have to make `__version__` work.

EUPS will use the release tag on the git repo so if you tag 0.1.0 that will be the version reported by eups.

Reply:
I believe that if we push a `0.1.0` tag into this repository, then `pkgautoversion` (and EUPS, etc) will pick up the version from that. Hence the latest weekly of astro_metadata_translator is `0.1.0-12-g03ebef7+5`, rather than `w.2020.25-stuff`.

My feeling is that :

- Spencer is right that there should be one source of truth for the version number;
- I prefer that being a git tag rather than something written into `setup.cfg` or into sconsUtils magic.

Thoughts? I'm sure there's best practice on generating a setuptools version from git, but I'll have to look up what it is...

Reply:
https://pypi.org/project/setuptools-scm/ is here to help! It's pretty good, and I support using it instead of `setup.cfg` or `sconsUtils`.

Reply:
Yes. I think SQuaRE use setuptools-scm for their pip deployments. We still need to make sure that the pip installable version reports a version number in `__version__` but only if you use sconsUtils do we mandate that method.